### PR TITLE
[DD-390] Add testing plan

### PR DESCRIPTION
**Pull request by chantelle11**
Ticket: [DD-390](https://circleci.atlassian.net/browse/DD-390) 

# Description
Until more automated tests are made we need a manual testing strategy going forward. 

# Reasons
We are getting incident reports of merging PR’s to prod breaking functionality. In an effort to ensure that the code does not break any other features we want to make a manual testing plan. 

While we are still building our automated testing suites we will need a manual testing plan that includes everything that needs to be tested. We can slowly move things from manual tests to automated tests while they are written.

### CODE REVIEW

File: docs/testing-plan.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+# Goal of testing plan
+Manual testing is expensive and we want to create a testing plan to focus on critical parts of the software while automated tests are being developed. 
+
+Some manual tests may be expensive to automate but we should aim to limit this list as much as possible and create as much test coverage to be confident in our deploys.
+
+When testing manual testing we need to keep in mind still both unit tests and integration tests.
```
Comment (tshen815) [inline docs/testing-plan.md:6]:
```suggestion
When testing manual testing we need to keep in mind still both unit tests and integration tests.
```
This seems a bit repetitive, what about something like:
```
When testing manually, we need to keep in mind both unit and integration tests.

Comment (tshen815) [inline docs/testing-plan.md:6]: This line seems a bit repetitive. What about something like:
```
When testing manually, we need to keep in mind both unit and integration tests.
```

File: docs/testing-plan.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+Unit tests: individual units or components of a software are tested. 
+Example: A button. Does it look like the mock up? Does it click event work as expected?
```
Comment (tshen815) [inline docs/testing-plan.md:9]:
```suggestion
Example: A button. Does it look like the mock up? Does the click event work as expected? 
```

Comment (tshen815) [reply docs/testing-plan.md:9]: Also, we should decide if we want to capitalize after the colon or not. Im not the best when it comes to grammar so Im not sure what the rule is 😅. In the other examples you arent capitalizing after the colon.

File: docs/testing-plan.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+Some manual tests may be expensive to automate but we should aim to limit this list as much as possible and create as much test coverage to be confident in our deploys.
+
+When testing manual testing we need to keep in mind still both unit tests and integration tests.
+
+Unit tests: individual units or components of a software are tested. 
+Example: A button. Does it look like the mock up? Does it click event work as expected? 
+
+Integrated tests: is the phase in software testing in which individual software modules are combined and tested as a group.How the user can interact with the application throughout an entire workflow.
+Example: A user navigates to a website. The user logs in. The user changes account name. The user logs out.
```
Comment (tshen815) [inline docs/testing-plan.md:12]: the phase in software testing in which individual software modules are combined and tested as a group. The user can interact with the application throughout an entire workflow.
Example: A user navigates to a website, logs in, changes the account name, and finally logs out.

Comment (tshen815) [inline docs/testing-plan.md:12]:
```suggestion
Integrated tests: the phase in software testing in which individual software modules are combined and tested as a group. The user can interact with the application throughout an entire workflow.
Example: A user navigates to a website, logs in, changes the account name, and finally logs out.
```

File: docs/testing-plan.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+- Determine test Deliverables
+
+## 2.2 Tasks
+- testing 
+- problem reporting
+- re-testing
+
+# 3.0 Tactics
+## Testing
+We need to test the existing interfaces. Follow the how to test instructions in the github pull request. If test instructions are not present check the README file in the main application folder on instructions how to run the application locally.
```
Comment (tshen815) [inline docs/testing-plan.md:36]:
```suggestion
We need to test the existing interfaces. Please refer to the **How to Test** instructions in the github pull request. If test instructions are not present, check the README file in the main application folder on instructions for how to run the application locally. 
```

Comment (tshen815) [reply docs/testing-plan.md:36]: It doesnt show in the comment, but maybe **How to Test** should be bolded?

File: docs/testing-plan.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+## 2.2 Tasks
+- testing 
+- problem reporting
+- re-testing
+
+# 3.0 Tactics
+## Testing
+We need to test the existing interfaces. Follow the how to test instructions in the github pull request. If test instructions are not present check the README file in the main application folder on instructions how to run the application locally. 
+
+Testing should include going over design reviews. Ensuring the pull request is in line with the mock-ups provided.
```
Comment (tshen815) [inline docs/testing-plan.md:38]:
```suggestion
Testing should include design reviews to ensure the pull request is in line with the mock-ups provided. 
```

File: docs/testing-plan.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+- re-testing
+
+# 3.0 Tactics
+## Testing
+We need to test the existing interfaces. Follow the how to test instructions in the github pull request. If test instructions are not present check the README file in the main application folder on instructions how to run the application locally. 
+
+Testing should include going over design reviews. Ensuring the pull request is in line with the mock-ups provided. 
+
+## Problem reporting
+If an issue is found well testing the pull request there are three ways to leave messages in github.
```
Comment (tshen815) [inline docs/testing-plan.md:41]:
```suggestion
If an issue is found while testing the pull request, there are three ways to leave a message in github:
```

File: docs/testing-plan.md (hunk 1 above)
Comment (rsylvian) [reply docs/testing-plan.md:6]: I also don't think we need to get into explaining what unit tests and e2e tests are. If we want this document to be helpful we should keep it as short as possible in my opinion.

File: docs/testing-plan.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+- Automated Regression Testing
+
+- Beta Testing
+This should take place when we are reviewing the PRs and spin up the application locally.
+
+# 5.0 Hardware Requirements
+Currently we only support spinning up development enviroments on Mac books.
+
+# 6.0 Features to Be Tested
+Note as this application is constnatly being iterated on this features to be tested list should always be reviewed and edited based on features added / removed or updated.
```
Comment (rsylvian) [inline docs/testing-plan.md:76]:
```suggestion
Note as this application is constantly being iterated on this features to be tested list should always be reviewed and edited based on features added / removed or updated.
```

Is it missing a word? I am having trouble to follow this sentence.

File: docs/testing-plan.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+Remember to test all features in both English and Japanese version of the Docs site.
+
+## Accessibility testing
+- Can you search in algolia?
+- Are the tabs in search toggleable?
+- Can you tab to `Skip to Content`?
+
+## Header
+- CircleCi logo that when clicked takes you to docs homepage
```
Comment (rsylvian) [inline docs/testing-plan.md:88]:
```suggestion
- CircleCi logo takes you to the homepage when clicked
```

File: docs/testing-plan.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+- Search bar (See search bar below more details)
+- Globe icon that opens to langauge selector (See language selector below for more details)
```
Comment (rsylvian) [inline docs/testing-plan.md:90]:
```suggestion
- Clicking the globe icon opens the language selector (See language selector below for more details)
```

File: docs/testing-plan.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+- Link to circleci.com
```
Comment (rsylvian) [inline docs/testing-plan.md:91]:
```suggestion
- Link to `circleci.com` is displayed
```

File: docs/testing-plan.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+- Button to `Go to application` or `Sign up`
```
Comment (rsylvian) [inline docs/testing-plan.md:92]:
```suggestion
- Button `Go to application` or `Sign up` are displayed
```

File: docs/testing-plan.md [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+
+## Search bar
+- When typing inside of search bar then results a results page should appear. The url should now contain a query string of the word used. 3 tabs should appear
```
Comment (rsylvian) [inline docs/testing-plan.md:95]:
```suggestion
- When searching, results should appear. The URL should also contain a query string of the words used. 3 tabs should appear:
```

File: docs/testing-plan.md [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+1. Documention - results found in /docs
+2. Orbs - Results found in the developer/orb site
+3. Convenience Images - Results found in developer/orb
+- Pressing cancel should close search bar results 
+- Pressing `X` should clear search results and query string from url
+
+## Side nav
```
Comment (rsylvian) [inline docs/testing-plan.md:102]:
```suggestion
## Sidebar
```

File: docs/testing-plan.md [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+- Contains a list to all different resource available on the Docs site
+- There should be top level sections that toggle between open and closed when clicked on revealing child lists
```
Comment (rsylvian) [inline docs/testing-plan.md:104]:
```suggestion
- There should be top-level sections that toggle between opened and closed. Toggling it should reveal child lists 
```

File: docs/testing-plan.md [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+1. Documention - results found in /docs
+2. Orbs - Results found in the developer/orb site
+3. Convenience Images - Results found in developer/orb
+- Pressing cancel should close search bar results 
+- Pressing `X` should clear search results and query string from url
+
+## Side nav
+- Contains a list to all different resource available on the Docs site
+- There should be top level sections that toggle between open and closed when clicked on revealing child lists 
+- The side nav should ensure the parent category is opened and the child side page header is highlighted in the list for the associated page that the user is currently viewing
```
Comment (rsylvian) [inline docs/testing-plan.md:105]:
```suggestion
- The sidebar should ensure the parent category is opened and the child side page header is highlighted in the list for the associated page that the user is currently viewing
```

I think that sentence might need a rework. A little hard to follow IMO

File: docs/testing-plan.md [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+- When an item is clicked on the side nav it should navigate the user to the associated page, and highlight the item should be highlighted
```
Comment (rsylvian) [inline docs/testing-plan.md:106]:
```suggestion
- When an item is clicked in the sidebar, it should navigate the user to the associated page as well as, highlight the correct item
```

File: docs/testing-plan.md [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+
+## TOC
```
Comment (rsylvian) [inline docs/testing-plan.md:108]:
```suggestion
## Table of contents (TOC)
```

File: docs/testing-plan.md [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+- TOC may not appear on all pages
```
Comment (rsylvian) [inline docs/testing-plan.md:109]: Here we should give examples of pages.

File: docs/testing-plan.md [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+- TOC contains a list of headers that are on the page and helpful resources
```
Comment (rsylvian) [inline docs/testing-plan.md:110]: TOC and Helpful resources are two different things. Let's not mix both in the same section please :)

File: docs/testing-plan.md [hunk 20]
```diff
@@ -0,0 +1,130 @@
...
+- When a user clicks on a helpful resource the user should be navigated to that resource
+- Wehn a user clicked on an item in `On this page` then the user should be scrolled to that item on the page and the item should now be highlighted
```
Comment (rsylvian) [inline docs/testing-plan.md:112]:
```suggestion
- When a user clicked on an item in `On this page`, they should be scrolled to that item on the page and the item should now be highlighted
```

File: docs/testing-plan.md [hunk 21]
```diff
@@ -0,0 +1,130 @@
...
+
+## Language selector
+- Language selector allows a user to switch between English and Japanese
+
+## Screen size
+- The header will change depending on screen size. For example the circle ci home page may no longer be visible.
```
Comment (rsylvian) [inline docs/testing-plan.md:118]: Not sure what this means? Homepage is visible in mobile?

File: docs/testing-plan.md [hunk 22]
```diff
@@ -0,0 +1,130 @@
...
+## TOC
+- TOC may not appear on all pages
+- TOC contains a list of headers that are on the page and helpful resources
+- When a user clicks on a helpful resource the user should be navigated to that resource
+- Wehn a user clicked on an item in `On this page` then the user should be scrolled to that item on the page and the item should now be highlighted
+
+## Language selector
+- Language selector allows a user to switch between English and Japanese
+
+## Screen size
```
Comment (rsylvian) [inline docs/testing-plan.md:117]:
```suggestion
## Responsiveness 
```

File: docs/testing-plan.md [hunk 23]
```diff
@@ -0,0 +1,130 @@
...
+- The header will change depending on screen size. For example the circle ci home page may no longer be visible. 
+- The side nav moves from the left hand side to be used inside of a hamburger icon
```
Comment (rsylvian) [inline docs/testing-plan.md:119]:
```suggestion
- The sidebar moves from the left-hand side to be used inside of a hamburger icon
```

File: docs/testing-plan.md [hunk 24]
```diff
@@ -0,0 +1,130 @@
...
+-  TOC moves from right side of the screen to being embeded at the top of the document
```
Comment (rsylvian) [inline docs/testing-plan.md:120]:
```suggestion
- TOC moves from the right side of the screen to being embedded at the top of the document
```

File: docs/testing-plan.md [hunk 25]
```diff
@@ -0,0 +1,130 @@
...
+
+# 7.0 Features Not to Be Tested
+- circleci.com
+- sign up / signin
```
Comment (rsylvian) [inline docs/testing-plan.md:124]: I think we should at least make sure the buttons accurately reflect user status.

File: docs/testing-plan.md [hunk 26]
```diff
@@ -0,0 +1,130 @@
...
+- API v1 and API v2 can be skipped unless feature relates to them specifically
+
+# 8.0 Resources/Roles & Responsibilities
+Docs team: For content verification
+Japanese translation team: To ensure Japanese translation is accurate
```
Comment (rsylvian) [inline docs/testing-plan.md:129]:
```suggestion
Translation team: To ensure Japanese translations are accurate
```

**Review by rsylvian — CHANGES_REQUESTED** [review]
